client/src/components/modal-forms/AssignCohort.js

```diff
@@ -0,0 +1,143 @@
+
+export const AssignCohortForm = ({ initialValues = {}, onClose, onSubmit, participantId }) => {
+  // State: PSI List to fill the select field
+  const [psiList, setPSIList] = useState([]);
+  // State: Cohort list connected with the PSI
+  const [cohorts, setCohorts] = useState([]);
+  // State Cohort Map
+  const [cohortMapData, setCohortMapData] = useState({});
+  // State: Cached PSI data
+  const [cachedPSIData, setCachedPSIData] = useState([]);
+
+  // Show Error
+  const [showError, setShowError] = useState(false);
+
+  // UseEffect: Fetch PSI List
+  useEffect(() => {
+    fetchPSIData().then(({ psiMappedData, cohortMap, psiData }) => {
+      setPSIList(psiMappedData);
+      setCohortMapData(cohortMap);
+      setCachedPSIData(psiData);
+    });
+  }, [setPSIList, setCohortMapData, setCachedPSIData]);
+
+  return (
+    <Box spacing={8} p={10}>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={ParticipantAssignCohortSchema}
+        onSubmit={(values) =>
+          postCohortAssignmentForm({
+            values,
+            participantId,
+            afterPostAction: onSubmit,
+            setShowError,
+            cachedPSIData,
+          })
+        }
+      >
+        {({ submitForm, values, setFieldValue }) => (
+          <FormikForm>
+            <Typography color={'primary'} variant={'h4'}>
+              {' '}
+              Assign Cohort
+            </Typography>
+            <hr />
+            <MuiAlert severity='warning'>
+              {
+                'This Participant has not been assigned a cohort. Please assign a cohort to track graduation.'
+              }
+            </MuiAlert>
+            <br />
+            <>
+              {showError && (
+                <MuiAlert severity='error'>
+                  {'Unable to assign cohort. Please try again later.'}
+                </MuiAlert>
+              )}
+            </>
+            <br />
+            <Box>
+              <Field
+                name='institute'
+                type='number'
+                component={RenderSelectField}
+                label='* Institute Name'
+                options={psiList}
+                onChange={({ target }) => {
+                  const { value } = target;
+                  setFieldValue('institute', value);
+                  setCohorts(cohortMapData[value]);
+                }}
+              />
+              <Field
+                name='cohort'
+                type='number'
+                component={RenderSelectField}
+                label='* Cohort'
+                options={cohorts}
+              />
+            </Box>
+            <Box mt={3}>
+              <Grid container spacing={2} justify='flex-end'>
+                <Grid item>
+                  <Button onClick={onClose} color='default' text='Cancel' />
+                </Grid>
+                <Grid item>
+                  <Button onClick={submitForm} variant='contained' color='primary' text='Submit' />
```

Comment:
does "Submit" match our other in-app styling? the designs have "Save changes" but I know we wanted to stick to consistency over matching designs for the time being

Reply:
Label updated